repose-aggregator/components/filters/openstack-identity-v3/src/test/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3HandlerTest.scala

```diff
@@ -291,35 +291,56 @@ class OpenStackIdentityV3HandlerTest extends FunSpec with BeforeAndAfter with Ma
 
   describe("writeProjectHeader") {
     val writeProjectHeader = PrivateMethod[Unit]('writeProjectHeader)
-    val filterDirector = mock[FilterDirector]
     val headerManager = mock[HeaderManager]
-    when(filterDirector.requestHeaderManager()).thenReturn(headerManager)
     val roles = List(Role(null, null, Option("12345"), null, null), Role(null, null, Option("67890"), null, null))
 
     it("should only provide the url project when the flag says to not write all") {
-      val randomness = java.util.UUID.randomUUID.toString
-      identityV3Handler invokePrivate writeProjectHeader(randomness, roles, false, filterDirector)
-      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(randomness))
+      val defaultPid = java.util.UUID.randomUUID.toString
+      val uriPid = java.util.UUID.randomUUID.toString
+      identityV3Handler invokePrivate writeProjectHeader(Some(defaultPid), roles, Some(uriPid), false, false, headerManager)
+      verify(headerManager).appendHeader(org.mockito.Matchers.eq("X-Project-ID"), org.mockito.Matchers.eq(uriPid))
```

Comment:
So that doesn't quite work, but there is a better alternative.

import org.mockito.Matchers.{eq => mockitoEq}